include/ccmath/math/expo/log.hpp

```diff
@@ -30,33 +32,30 @@ namespace ccm
 	template <typename T, std::enable_if_t<!std::is_integral_v<T>, bool> = true>
 	constexpr T log(const T num) noexcept
 	{
-#if defined(__GNUC__) && (__GNUC__ > 6 || (__GNUC__ == 6 && __GNUC_MINOR__ >= 1)) && !defined(__clang__)
-		if constexpr (std::is_same_v<T, float>) { return __builtin_logf(num); }
-		if constexpr (std::is_same_v<T, double>) { return __builtin_log(num); }
-		if constexpr (std::is_same_v<T, long double>) { return __builtin_logl(num); }
-		return static_cast<T>(__builtin_logl(num));
-#else
-		// If the number is 1, return +0.
-		if (num == static_cast<T>(1)) { return static_cast<T>(0); }
+		if constexpr (ccm::builtin::has_constexpr_log<T>) { return ccm::builtin::log(num); }
+		else
+		{
+			// If the number is 1, return +0.
+			if (num == static_cast<T>(1)) { return static_cast<T>(0); }
```

Comment:
I think you can use T{1} and T{0} instead of static casts, I might be wrong but you tell me

Reply:
From my understanding structurally they are the same, but from a C++ perspective in being explicit I'd say semantically the static_cast would be better here as it's very precise in the outcome. Also T{} is more like a C style cast in my opinion even though it is not. So I'd say I'd us rather prefer the usage of static_cast instead.

Reply:
You're right, I just checked and T{} assumes that this type supports brace initialization. Static cast would be explicit way of converting value to that type